### PR TITLE
Update_config script tests added and fixed - Closes #116

### LINKED ISSUE #116: Partial custom config migration during node upgrade
### Expected behavior
When a node is upgraded all the custom config should be migrated and ones which are not supported should be shown as `warning` or `info`.

**Note: Node upgrade has a recurring issue with one or the other config fields, this time while we fix the issue, whoever is working on it and approving the PR please make sure to add tests by scenario or whichever is convenient, but the tests should cover all possible scenarios to avoid such recurring issues, Danke!.**

### Actual behavior
Upgrading a node with version `2.0.0` to `2.0.x` does not migrate some of the custom config fields like `logger` under `components`.
### Steps to reproduce
- Deploy 2.0.0 node
- Create a custom config(2.0.0 node) with
```
{
	"components": {
		"logger": {
			"fileLogLevel": "info"
		}
	}
}
```
- Upgrade to `2.0.1` version
- Observe the `config.json`, it will not include the custom `logger` config
### Which version(s) does this affect? (Environment, OS, etc...)
2.0.1, Binary release

### PR DESCRIPTION
**Pull request by michielmulders**
### What was the problem?
Few config changes were not correctly incorporated.
Not enough tests.

### How did I fix it?
Created `integration` tests (moved `unit` to integration).
Fixes in `history.js` file for config
Added tests covering most scenarios migrating 1.6.0 to 2.0.0 for devnet/mainnet/testnet config.

### How to test it?
`npm run test`

### Review checklist

* The PR resolves #116 
* All new code is covered with integration tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: scripts/update_config.js [hunk 1]
```diff
@@ -321,7 +325,7 @@ history.version('2.0.0-rc.0', version => {
 		);
 		config = moveElement(
 			config,
-			'modules.chain.broadcasts.broadcastLimit',
```
Comment (shuse2) [inline scripts/update_config.js]: Is this correct? I think broadcastLimit is already moved to chain in `config = moveElement(config, 'broadcasts', 'modules.chain.broadcasts');`

Comment (michielmulders) [reply scripts/update_config.js]: Yes, but for the next change, the broadcast limit is moved to `modules.network.emitPeerLimit` for `2.0`

File: test/integration/scripts/update_config.js [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+		// Act
+		const updatedConfig = getUpdatedConfig(outputPath, inputPath, '1.6.0', '2.0.0');
+
+		// Assert
+		expect(updatedConfig.components.storage.host).to.eql(undefined);
+		expect(updatedConfig.components.storage.port).to.eql(undefined);
+	});
+});
+
+describe('migrate from 1.6.0 to 2.0.0 for testnet', () => {
```
Comment (ManuGowda) [inline test/integration/scripts/update_config.js]: this description is similar to above one, I hope you intended to upgrade and test devnet here, can you review this and update accordingly?

File: scripts/update_config.js (hunk 1 above)
Comment (michielmulders) [reply scripts/update_config.js]: `framework/src/modules/network/defaults/config.js:75` here is it now

File: test/integration/scripts/update_config.js [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+	afterEach(async () => {
+		fs.unlinkSync(outputPath);
+	});
+
+	after(async () => {
+		fs.unlinkSync(inputPath);
+	});
+
+	it('should detect custom config for wsPort', async () => {
```
Comment (ManuGowda) [inline test/integration/scripts/update_config.js]: please add scenarios related to forging(access, whitelist, force, delegates, password) sections, SSL and network(seedpeers, blacklist).

Comment (michielmulders) [reply test/integration/scripts/update_config.js]: Added those scenarios!

File: scripts/update_config.js [hunk 4]
```diff
@@ -440,10 +449,11 @@ history.migrate(
 			console.info(`\nWriting configuration file to ${program.output}`);
 			fs.writeFileSync(
 				program.output,
-				JSON.stringify(customConfig, null, '\t')
+				JSON.stringify(customConfig, null, '\t'),
+				{ flag: 'wx' }
 			);
 		} else {
-			console.info('\n\n------------ OUTPUT -------------');
+			console.infso('\n\n------------ OUTPUT -------------');
```
Comment (ManuGowda) [inline scripts/update_config.js]: typo?

File: test/fixtures/config_2.0.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+	"components": {
+		"logger": {
+			"fileLogLevel": "info",
+            "consoleLogLevel": "debug"
```
Comment (ManuGowda) [inline test/fixtures/config_2.0.json]: formatting has gone wrong

File: test/integration/scripts/update_config.js [hunk 6]
```diff
@@ -0,0 +1,636 @@
+/*
+ * Copyright © 2018 Lisk Foundation
```
Comment (ManuGowda) [inline test/integration/scripts/update_config.js]: 2019

File: test/integration/scripts/update_config.js [hunk 7]
```diff
@@ -0,0 +1,636 @@
...
+		config.peers = {
+			list: [
+				{
+					ip: 'mynode.ip.net',
+					wsPort: '5000',
+				}
+			],
+			access: {
+				blackList: [
+					'11c8b3d6a9e418fff20ef58383260bcd04799db150612d4ff6eb399bcd07f111'
```
Comment (ManuGowda) [inline test/integration/scripts/update_config.js]: this should be an ip address know?

Comment (ManuGowda) [reply test/integration/scripts/update_config.js]: if we migrate any value, then the application might crash if the error is not handled.

**Comment by fchavant** [comment]
@michielmulders where do those test run? (Jenkins URL?)

File: test/integration/scripts/update_config.js [hunk 8]
```diff
@@ -0,0 +1,636 @@
...
+
+		// Act
+		const updatedConfig = getUpdatedConfig(outputPath, inputPath, '1.6.0', '2.0.0');
+
+		// Assert
+		expect(updatedConfig.components.storage.host).to.eql(config.db.host);
+		expect(updatedConfig.components.storage.port).to.eql(config.db.port);
+	});
+
+	it('should detect custom config for db.host and db.port', async () => {
```
Comment (fchavant) [inline test/integration/scripts/update_config.js]: Wrong comment?

File: test/integration/scripts/update_config.js [hunk 9]
```diff
@@ -0,0 +1,636 @@
...
+		config.peers = {
+			list: [
+				{
+					ip: 'mynode.ip.net',
+					wsPort: '5000',
+				}
+			],
+			access: {
+				blackList: [
+					'11c8b3d6a9e418fff20ef58383260bcd04799db150612d4ff6eb399bcd07f111'
```
Comment (fchavant) [inline test/integration/scripts/update_config.js]: Shouldn't this be an IP?

File: test/integration/scripts/update_config.js [hunk 10]
```diff
@@ -0,0 +1,636 @@
...
+		expect(updatedConfig.modules.http_api.ssl.options).to.eql(config.api.ssl.options);
+	});
+
+	it('should detect custom config for forging', async () => {
+		// Arrange
+		const config = cloneDeep(baseConfig);
+		config.forging = {
+			force: true,
+			delegates: [
+				'47c8b3d6a9e418f0920ef58383260bcd04799db150612d4ff6eb399bcd07f216',
```
Comment (fchavant) [inline test/integration/scripts/update_config.js]: This does not look like the right format, see https://github.com/LiskHQ/lisk-core/blob/development/config/devnet/config.json#L34

File: scripts/update_config.js [hunk 11]
```diff
@@ -440,10 +449,11 @@ history.migrate(
 			console.info(`\nWriting configuration file to ${program.output}`);
 			fs.writeFileSync(
 				program.output,
-				JSON.stringify(customConfig, null, '\t')
+				JSON.stringify(customConfig, null, '\t'),
+				{ flag: 'wx' }
```
[the remark this answers is not in the document]
Comment (michielmulders) [reply scripts/update_config.js]: Ah you are right, just writable!

**Comment by ManuGowda** [comment]
@michielmulders @fchavant made a good point, we need to integrate these tests somewhere, and make sure when we have new changes to update config, the tests are covered and all green.

**Pull request merged**